### PR TITLE
doc: Update Windows build instructions

### PR DESCRIPTION
**Pull request by barton2526**
This ports four commits from upstream:
- Remove an outdated comment
- Add a note to use `-j` for parallel builds
- Improve note on choosing POSIX
- Remove WSL install instructions and point to Microsoft doc instead

### CODE REVIEW

File: doc/build-windows.md [hunk 1]
```diff
@@ -5,11 +5,9 @@ Below are some notes on how to build Gridcoin for Windows.
...
 The options known to work for building Gridcoin on Windows are:
 
-* On Linux, using the [Mingw-w64](https://mingw-w64.org/doku.php) cross compiler tool chain. Ubuntu Bionic 18.04 is required
-and is the platform used to build the Gridcoin Windows release binaries.
-* On Windows, using [Windows
-Subsystem for Linux (WSL)](https://docs.microsoft.com/windows/wsl/about) and the Mingw-w64 cross compiler tool chain.
-* On Windows, using a native compiler tool chain such as [Visual Studio](https://www.visualstudio.com).
+* On Linux, using the [Mingw-w64](https://www.mingw-w64.org/) cross compiler tool chain.
+* On Windows, using [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/windows/wsl/about) and Mingw-w64.
+* On Windows, using [Microsoft Visual Studio](https://www.visualstudio.com).
```
Comment (jamescowens) [inline doc/build-windows.md]: Have we actually tried MVS recently? I haven't.

Comment (barton2526) [reply doc/build-windows.md]: I can easily remove that line

Comment (jamescowens) [reply doc/build-windows.md]: I think we should, don't you? I know @cyrossignol was going to try and get MVS to work, but noone has time to fool with it, and why would we want to anyway... right?

**Review by jamescowens — APPROVED** [review]
utACK

**Pull request merged**